File: curso_c_cpp/c/04_operadores/ejemplos/logicos.c
/*
 * Operadores relacionales y logicos. En C: 1 = verdadero, 0 = falso.
 */
#include <stdio.h>

int main(void) {
    int edad = 20;
    int tieneEntrada = 1;

    printf("edad >= 18: %d\n", edad >= 18);            // 1
    printf("edad == 30: %d\n", edad == 30);            // 0
    printf("mayor Y con entrada: %d\n", edad >= 18 && tieneEntrada);  // 1
    printf("menor O con entrada: %d\n", edad < 18 || tieneEntrada);   // 1
    printf("NO tiene entrada: %d\n", !tieneEntrada);   // 0
    return 0;
}

File: curso_java/04_operadores/ejemplos/Logicos.java
/*
 * Operadores relacionales y logicos: en Java el resultado es boolean.
 */
public class Logicos {
    public static void main(String[] args) {
        int edad = 20;
        boolean tieneEntrada = true;

        System.out.println("edad >= 18: " + (edad >= 18));            // true
        System.out.println("edad == 30: " + (edad == 30));            // false
        System.out.println("mayor Y con entrada: " + (edad >= 18 && tieneEntrada));  // true
        System.out.println("menor O con entrada: " + (edad < 18 || tieneEntrada));   // true
        System.out.println("NO tiene entrada: " + (!tieneEntrada));   // false
    }
}
